File: native/mixdog-graph/tests/fixtures/parity/csharp/Sample.cs
// Parity fixture: declaration and import shapes the graph reports for C#.
using System;
using System.Collections.Generic;
using static System.Math;
using Alias = System.Text.StringBuilder;

namespace Acme.Sample {
  public interface IListener {
    void OnEvent(string message);
  }

  public enum Mode {
    Fast,
    Slow
  }

  public struct Point {
    public int X;
    public int Y;

    public int Sum() => X + Y;
  }

  public record Pair(string Left, string Right);

  public class Store : IListener {
    private readonly string name;

    public Store(string name) {
      this.name = name;
    }

    public void OnEvent(string message) {
      Console.WriteLine(message);
    }

    public string Read(string key) {
      string Local(string value) => value.Trim();
      var builder = new Alias();
      builder.Append(Local(key));
      return $"{name}:{builder}";
    }

    public static double Root(double value) => Sqrt(value);

    private class Inner {
      public List<string> Items() => new List<string>();
    }
  }
}

File: native/mixdog-graph/tests/fixtures/parity/cpp/sample.cpp
// Parity fixture: declaration and import shapes the graph reports for C++.
#include <string>
#include <vector>
#include "sample.hpp"

namespace acme {

struct Point {
  int x;
  int y;

  int sum() const { return x + y; }
};

class Store {
 public:
  explicit Store(std::string name) : name_(std::move(name)) {}

  std::string read(const std::string &key) const { return name_ + ":" + key; }

  static Store build(const std::string &name);

 private:
  std::string name_;
};

Store Store::build(const std::string &name) { return Store(name); }

std::vector<std::string> *collect(const Store &store) {
  auto *items = new std::vector<std::string>();
  items->push_back(store.read("k"));
  return items;
}

int main() {
  Point point{1, 2};
  return point.sum();
}

}  // namespace acme
